tensorflow/compiler/jit/mark_for_compilation_pass.cc

```diff
@@ -1076,6 +1076,86 @@ StatusOr<bool> IsIdentityDrivingConstsInLoop(Node* node) {
+      "BitwiseOr", "BitwiseXor", "LeftShift", "RightShift", "LogicalAnd",
+      "LogicalOr", "Mod", "Maximum", "Minimum", "RealDiv", "ReciprocalGrad",
+      "RsqrtGrad", "SqrtGrad", "TruncateDiv", "TruncateMod", "Equal",
+      "NotEqual", "Greater", "GreaterEqual", "Less", "LessEqual", "SigmoidGrad",
+      "SoftplusGrad", "SoftsignGrad", "TanhGrad", "Pow", "SquaredDifference",
+      "ApproximateEqual",
+      // Others
+      "AddN", "Bitcast", "Cast", "ClipByValue", "Const", "Empty", "Identity",
+      "IdentityN", "Relu", "Relu6", "ReluGrad", "Relu6Grad", "LeakyReluGrad",
+      "Elu", "EluGrad", "Selu", "SeluGrad", "Select", "SelectV2", "Transpose",
+      "ConjugateTranspose", "_UnaryOpsComposition",
+      // The following 4 operations are converted to identity
+      "PlaceholderWithDefault", "PreventGradient", "StopGradient", "Snapshot"}},
+    // clang-format off
+    {"RED",
+     {"All", "Any", "Min", "Max", "Mean", "Prod", "Sum"}},
+    // clang-format on
+    {"PWRED",
+     {"ArgMax", "ArgMin", "DiagPart", "Softmax",
+      "SparseSoftmaxCrossEntropyWithLogits", "LogSoftmax"}},
+    {"REDUCEWINDOW",
+     {"ArgMax", "ArgMin", "DiagPart", "Softmax",
+      "SparseSoftmaxCrossEntropyWithLogits", "LogSoftmax"}},
+    {"REDUCEWINDOWPW", {"BiasAddGrad", "LRN", "LRNGrad"}},
+    {"BN",
+     {"FusedBatchNorm", "FusedBatchNormV2", "FusedBatchNormV3",
+      "_FusedBatchNormEx", "FusedBatchNormGrad", "FusedBatchNormGradV2",
+      "FusedBatchNormGradV3"}},
+    {"SORT", {"TopKV2"}}, // XLA version much faster then TF version.
+    {"SMALL",
+     // clang-format off
+     {"BroadcastTo", "ExpandDims", "Fill", "NoOp",
+      "Range", "Rank", "Reshape", "Shape", "ShapeN", "Size", "Squeeze",
+      "Transpose", "ZerosLike", "OnesLike", "BiasAdd" /*PW + Broadcast*/,
+      "BroadcastArgs", "BroadcastGradientArgs", "OneHot", "Concat", "ConcatV2",
+      "ConcatOffset", "Const", "MirrorPad", "Pack", "Pad", "PadV2", "Reverse",
+      "ReverseV2", "ReverseSequence", "Slice", "Split", "SplitV",
+      "StridedSlice", "StridedSliceGrad", "ResourceStridedSliceAssign",
+      "Tile", "Transpose", "InvertPermutation", "Unpack"}}};
+// clang-format on
+
+std::unique_ptr<absl::flat_hash_set<string>> GetWhitelist() {
+  MarkForCompilationPassFlags* flags = GetMarkForCompilationPassFlags();
+  auto whitelist = absl::WrapUnique(new absl::flat_hash_set<string>());
+
+  for (auto s : absl::StrSplit(flags->tf_xla_ops_to_cluster, ",")) {
+    if (s == "FUSIBLE") {
+      for (auto pair : whitelist_table) {
+        whitelist->insert(pair.second.begin(), pair.second.end());
+      }
+    } else if (whitelist_table.contains(s)) {
+      auto v = whitelist_table.at(s);
+      whitelist->insert(v.begin(), v.end());
+    } else if (s.size() > 0) {
+      // Should be a user provided TF operation.
+      whitelist->insert(string(s));
+    }
+  }
+
+  if (VLOG_IS_ON(2) && whitelist->size() > 0) {
```

Comment:
Why not just sort on all codepaths, then this entire branch becomes `VLOG(2) << ...`

Reply:
To remove the time for the sort when we do not print.
I tried to change the container to the sorted container absl::btree_set to remove the need for a sort. But btree_set isn't available in my version of absl. I'm not sure it is a good reason to bump TF absl required version and never changed TF dependency version.

If I put a VLOG(2) on all code path, this will make the output verbose even when this feature isn't used. Are you suggesting doing that? I wouldn't have a tendency of adding extra useless verbose output here. Do you see value to always print it even when not used?